### PR TITLE
backport: trivial 2024 07 23

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Batch of trivial backports

## How Has This Been Tested?
building; ran tests that were changed

## Breaking Changes
  _Please describe any breaking changes your code introduces_

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -2696,6 +2696,9 @@ class CoinsViewScanReserver
 
 static RPCHelpMan scantxoutset()
 {
+    // scriptPubKey corresponding to mainnet address 12cbQLTFMXRnSzktFkuoG3eHoMeFtpTu3S
```
Comment (knst) [inline src/rpc/blockchain.cpp:2699]: > 
  ~Note that the example descriptor used doesn't follow the developer guideline of using invalid bech32 addresses, as the RPC is not wallet-related and it's use-case is merely to look up state information (i.e. there is no danger of sending funds to a wrong address).~ For the sake of simplicity, the raw descriptor for an early coinbase payout address (block 9) is taken, i.e. it yields results even at an early stage of IBD. Happy to change that though if there are other suggestions.


Should it be a valid dash address?

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -794,7 +794,7 @@ static RPCHelpMan getblockfrompeer()
     ChainstateManager& chainman = EnsureChainman(node);
     PeerManager& peerman = EnsurePeerman(node);
 
-    const uint256& block_hash{ParseHashV(request.params[0], "block_hash")};
+    const uint256& block_hash{ParseHashV(request.params[0], "blockhash")};
```
Comment (knst) [inline src/rpc/blockchain.cpp:797]: is it breaking change? should we have release notes?

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp:2699]: meh? @UdjinM6

Comment (UdjinM6) [reply src/rpc/blockchain.cpp:2699]: yeah, why not.

```
> dash-cli deriveaddresses 'raw(76a91411b366edfc0a8b66feebae5c2e25a7b6a5d1cf3188ac)#fm24fxxy'
[
  "XcJSEb79KEeNbwMU7eE27aL5dhDwvjgBuH"
]
```

Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp:2699]:
```suggestion
    // scriptPubKey corresponding to mainnet address XcJSEb79KEeNbwMU7eE27aL5dhDwvjgBuH
```

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp:797]: I don't think this is breaking; it's only the name of the param, nothing more. No way this could break someone's flow

Comment (UdjinM6) [reply src/rpc/blockchain.cpp:797]: well, if someone is using named params it would be indeed a breaking change imo

Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp:797]: can you do named params with rpc?

Comment (UdjinM6) [reply src/rpc/blockchain.cpp:797]:
```
> dash-cli -named getblockfrompeer block_hash=000000000000002ef98b2f8679bf97c84e201de628e9cd59985c559e426c477d peer_id=1
{
  "warnings": "Block already downloaded"
}
```

Comment (UdjinM6) [reply src/rpc/blockchain.cpp:797]:
```
> dash-cli -named getblockfrompeer blockhash=000000000000002ef98b2f8679bf97c84e201de628e9cd59985c559e426c477d peer_id=1
error code: -8
error message:
Unknown named parameter blockhash
```
(on develop)

Comment (knst) [reply src/rpc/blockchain.cpp:797]: yes, of course, all of our functional tests do that. For example: https://github.com/dashpay/dash/blob/master/test/functional/feature_asset_locks.py#L207

Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp:797]: release notes pushed :)

File: doc/release-notes-24806.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+## RPC changes
+
+- `getblockfrompeer`, introduced in v21.0.0, named argument `block_hash` has been
+  renamed `blockhash` to be aligned with the rest of the codebase. This is a breaking
+  change if using named parameters. (bitcoin#24806)
```
Comment (knst) [inline doc/release-notes-24806.md:5]: nit: 
```suggestion
  change if using named parameters. (#6149)
```

**Review by knst — APPROVED** [review]
utACK e618ee10aaec67793f60cada579459b5fc4cc58b

File: doc/release-notes-24806.md (hunk 3 above)
Comment (UdjinM6) [reply doc/release-notes-24806.md:5]: #6149 can also point to bitcoin#6149 so let's be precise
```suggestion
  change if using named parameters. (dash#6149)
```

Comment (PastaPastaPasta) [reply doc/release-notes-24806.md]: I don't think it's important to specify dash here. It's in our release notes after all :)

Comment (UdjinM6) [reply doc/release-notes-24806.md]: Well, bitcoin devs add their repo prefix. I guess there might be a good reason for this.

**Review by UdjinM6 — APPROVED** [review]
utACK d4a728b60d7ca6708fd0e007f02f5e07be3c6678

**Pull request merged**